mvpa2/tests/test_generators.py

```diff
@@ -396,3 +397,79 @@ def is_sorted(x):
     assert_raises(ValueError, permutation, ds)
+
+
+def test_factorialpartitioner():
+    # Test against sifter and chainmap implemented in test_usecases
+    # -- code below copied from test_usecases --
+    # Let's simulate the beast -- 6 categories total groupped into 3
+    # super-ordinate, and actually without any 'superordinate' effect
+    # since subordinate categories independent
+    ds = normal_feature_dataset(nlabels=6,
+                                snr=100,   # pure signal! ;)
+                                perlabel=30,
+                                nfeatures=6,
+                                nonbogus_features=range(6),
+                                nchunks=5)
+    ds.sa['subord'] = ds.sa.targets.copy()
+    ds.sa['superord'] = ['super%d' % (int(i[1])%3,)
+                         for i in ds.targets]   # 3 superord categories
+    # let's override original targets just to be sure that we aren't relying on them
+    ds.targets[:] = 0
+
+    # let's make two other datasets to test later
+    # one superordinate category only
+    ds_1super = ds.copy()
+    ds_1super.sa['superord'] = ['super1' for i in ds_1super.targets]
+
+    # one superordinate category has only one subordinate
+    ds_unbalanced = ds.copy()
+    nsuper1 = np.sum(ds_unbalanced.sa.superord == 'super1')
+    mask_superord = ds_unbalanced.sa.superord == 'super1'
+    uniq_subord = np.unique(ds_unbalanced.sa.subord[mask_superord])
+    ds_unbalanced.sa.subord[mask_superord] = [uniq_subord[0] for i in range(nsuper1)]
+
+
+    npart = ChainNode([
+        ## so we split based on superord
+        NFoldPartitioner(len(ds.sa['superord'].unique),
+                         attr='subord'),
+        ## so it should select only those splits where we took 1 from
+        ## each of the superord categories leaving things in balance
+        Sifter([('partitions', 2),
+                ('superord',
+                 { 'uvalues': ds.sa['superord'].unique,
+                   'balanced': True})
+                ]),
+    ], space='partitions')
+
+    # now the new implementation
+    factpart = FactorialPartitioner(
+        NFoldPartitioner(attr='subord'),
+        attr='superord'
+    )
+
+    partitions_npart = [p.sa.partitions for p in npart.generate(ds)]
+    partitions_factpart = [p.sa.partitions for p in factpart.generate(ds)]
+
+    assert_array_equal(np.sort(partitions_npart), np.sort(partitions_factpart))
+
+    # now let's check it behaves correctly if we have only one superord class
+    nfold = NFoldPartitioner(attr='subord')
+    partitions_nfold = [p.sa.partitions for p in nfold.generate(ds_1super)]
+    partitions_factpart = [p.sa.partitions for p in factpart.generate(ds_1super)]
+    assert_array_equal(np.sort(partitions_nfold), np.sort(partitions_factpart))
+
+    # smoke test for unbalanced subord classes
+    partitions_factpart = [p.sa.partitions
+                           for p in factpart.generate(ds_unbalanced)]
```

Comment:
and what happens -- I guess warning is issued...  could you check if our assert_warnings works for this one? ;)
For that please do not use our ad-hoc mvpa2.base warning but `warnings.warn(msg, category=warnings.RuntimeWarning)` function, and here is an example from another place how to assert
```
from mvpa2.testing import assert_warnings
...
        with assert_warnings(
                [(DeprecationWarning,
                  "allowedtype option was deprecated in favor of constraints. "
                  "Adjust your code, provided value 'str' was ignored")]):
            DO_SOMETHIONG_WHICH_SPITS_WARNING
```

and what is the output? ;-) better to "hardcode" expectation here

Then I will become happy ;-)